tests/test_dmm.py

```diff
@@ -0,0 +1,167 @@
+# Copyright 2020 Pulser Development Team
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import cast
+from unittest.mock import patch
+
+import numpy as np
+import pytest
+
+from pulser.channels.dmm import DMM
+from pulser.register.base_register import BaseRegister
+from pulser.register.mappable_reg import MappableRegister
+from pulser.register.register_layout import RegisterLayout
+from pulser.register.weight_maps import DetuningMap
+
+
+@pytest.fixture
+def trap_coordinates():
+    return [[0, 0], [1, 0], [0, 1], [1, 1]]
+
+
+@pytest.fixture
+def layout(trap_coordinates) -> RegisterLayout:
+    return RegisterLayout(trap_coordinates)
+
+
+@pytest.fixture
+def register(layout: RegisterLayout) -> BaseRegister:
+    return layout.define_register(0, 1, 2, 3, qubit_ids=(0, 1, 2, 3))
+
+
+@pytest.fixture
+def map_reg(layout: RegisterLayout) -> MappableRegister:
+    return layout.make_mappable_register(4)
+
+
+def test_init(
+    layout: RegisterLayout, register: BaseRegister, map_reg: MappableRegister
+):
+    str_key = {"1": 1.0}
+    wrong_key = {4: 1.0}
+    bad_weights = {0: -1.0, 1: 1.0, 2: 1.0}
+    bad_sum = {0: 0.1, 2: 0.9, 3: 0.1}
+    det_map = {0: 0.7, 1: 0.3, 2: 0}
+    slm_map = {0: 1 / 3, 1: 1 / 3, 2: 1 / 3}
```

Comment:
I would define these closer to the check they are used in, but I realize it's a matter of personal preference